platform/core/lib/src/core_request.rs

```diff
@@ -0,0 +1,168 @@
+use crate::core::{get_instance, Core};
+use common::serialization::message_pack_serialize;
+use core_failure::CoreFailure;
+pub use feature_categories;
+use io::file_system::DefaultFileSystem;
+#[cfg(feature = "poly_rdf")]
+use poly_rdf::rdf::{SPARQLQuery, SPARQLUpdate};
+#[cfg(feature = "poly_rdf")]
+pub use poly_rdf::{
+    rdf::{QueryResults, QueryResultsFormat},
+    rdf_failure::RdfFailure,
+};
+use serde::Deserialize;
+#[allow(unused_imports)]
+use std::path::PathBuf;
+use user_session::TimeoutOption;
+#[cfg(target_os = "android")]
+use {
+    android_logger::Config,
+    log::{trace, Level},
+};
+
+type MessagePackBytes = Vec<u8>;
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LoadFeatureCategoriesArguments {
+    features_dir: String,
+    force_show: Vec<feature_categories::FeatureCategoryId>,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum CoreRequest {
+    LoadFeatureCategories {
+        args: LoadFeatureCategoriesArguments,
+    },
+    AppDidBecomeInactive,
+    IsUserSessionExpired,
+    SetUserSessionTimeout {
+        args: TimeoutOption,
+    },
+    GetUserSessionTimeoutOption,
+    GetUserSessionTimeoutOptionsConfig,
+    #[cfg(feature = "poly_rdf")]
+    ExecuteRdfQuery {
+        args: String,
+    },
+    #[cfg(feature = "poly_rdf")]
+    ExecuteRdfUpdate {
+        args: String,
+    },
+}
+
+pub fn execute_request(request: CoreRequest) -> MessagePackBytes {
```

Comment:
This function returns `MessagePackBytes` instead of `impl Serializable`, as Rust does not have, yet, complete support for `Existentials`, nor it is possible to create a `Trait Object` from a trait that has a function with generic parameters(which is the case for Serializable). Thus all of the response types are collapsed in the raw MessagePackBytes.